R/dictionary.R

```diff
@@ -280,3 +292,35 @@ vec_split <- function(x, by) {
 
   new_data_frame(list(key = keys, val = vals), n = vec_size(keys))
 }
+
+#' @export
+vec_split_old <- function(x, by) {
+
+  if (vec_size(x) != vec_size(by)) {
+    abort("`x` and `by` must have same size")
+  }
+
+  idx <- vec_duplicate_split_old(by)
+
+  keys <- vec_slice(by, map_int(idx, function(x) {x[1]}))
+
+  vals <- new_list_of(map(idx, vec_slice, x = x), vec_type(x))
+
+  new_data_frame(list(key = keys, val = vals), n = vec_size(keys))
+}
+
+#' @export
+vec_split_new <- function(x, by) {
+
+  if (vec_size(x) != vec_size(by)) {
+    abort("`x` and `by` must have same size")
+  }
+
+  idx <- vec_duplicate_split_new(by)
+
+  keys <- vec_slice(by, map_int(idx, function(x) {x[1]}))
+
+  vals <- new_list_of(map(idx, vec_slice, x = x), vec_type(x))
```

Comment:
I'd pull the map out on to it's own line for clarity

Reply:
✅